File: Lista Prof. Renan/Exercicios Lista 1/Execícios C_C++/C/André/OrdVet.cpp
//André Fernandes Gonçalves
//Matrícula 114038073

#include <stdio.h>
#include <conio.h>

int main(){
	
	int n, i, troca = 1, vetor[20], aux;
	
	printf("\n Este programa organiza um conjunto de no maximo vinte inteiros em ordem crescente. \n");
	printf("\n Insira o numero de elementos. \n");
	scanf("%d", &n);
	printf("\n Insira os valores. \n");
	for(i=0; i<n; i++){
		scanf("%d", &vetor[i]);
	}
	
	while (troca > 0){
		troca = 0; //Significa que ainda não houve troca de posição no vetor
		for (i=0; i<n-1; i++){
			if (vetor[i] > vetor[i+1]){
				aux = vetor [i];
				vetor[i] = vetor[i+1];
				vetor[i+1] = aux;
				troca = 1; //Houve troca
			}
		}
	}
	printf("\n O conjunto de numeros organizado em ordem crescente: \n");
	for (i=0; i<n; i++){
		printf("%2d", vetor[i]);
	}
	getch();
	return(0);	
}

File: Lista Prof. Renan/Exercicios Lista 1/FORTRAN/OrdVet.f90
!====================================================================
!André Fernandes Gonçalves
!Matrícula 114038073
!Linguagens de Programação para Engenharia Elétrica
!====================================================================
PROGRAM OrdVet
IMPLICIT NONE
!====================================================================
!Entrada de dados

INTEGER V(20), i, n, Aux, trocas, k
PRINT*,"Insira o numero de elementos de seu vetor"
READ(*,*)n
PRINT*,"Insira os elementos de seu vetor"
READ(*,*)(V(i),i=1,n)
!====================================================================
!Processamento

k=(n-1)
trocas=1
DO WHILE(trocas.GT.0)
    trocas=0
    DO i=1,k
        IF(V(i).GT.V(i+1))THEN
            Aux=V(i)
            V(i)=V(i+1)
            V(i+1)=Aux
            trocas=1
        END IF
    END DO
    k=k-1
END DO

!====================================================================
!Saída de dados

PRINT*,"O seu vetor ordenado em ordem crescente"
PRINT*,(V(i),i=1,n)

!====================================================================
STOP
END
